### ISSUE TITLE
[Spec]  Shell Flyout Footer Template

### ISSUE THREAD

**Issue by davidortinau** [xamarin/Xamarin.Forms#6292]
# Flyout Footer Template

Add a template for the footer of the flyout menu just like you have for the header. 

* The footer should be fixed to the bottom of the flyout. 
* The menu list should not go behind it and obscure the menu items.
* The footer should take any height.

```
<Shell.FlyoutFooterTemplate>
    <DataTemplate>
        <Grid>ContentHere</Grid>
    </DataTemplate>
</Shell.FlyoutFooterTemplate>
```

# API

Shell.FlyoutFooterTemplate takes a DataTemplate
 
# Difficulty : Low

**Comment by ahoefling** [comment]
I started an implementation of this for Andoird and I was able to get it working. I need to add unit tests and do some more testing.

Should I submit the PR with just the android implementation or would it be better to submit it with both iOS and Android?

![image](https://user-images.githubusercontent.com/17751436/58970204-ee9aca80-8786-11e9-9253-bfe916c73c95.png)

**Comment by PureWeen** [comment]
Yea! Submit a PR if you can

I have a PR over here
https://github.com/xamarin/Xamarin.Forms/pull/6383

where I'm working on some changes with the flyout measurements so we'll need to reconcile the two. But either way submit away!!!

**Comment by ahoefling** [comment]
Perfect! I'll submit the PR tomorrow morning.

It took me most of the night to get the bugs worked out with it but I have it working. I was running into a problem where the `RecyclerView` kept rendering behind the footer (red) and I wasn't able to adjust the margin. I ended up calculating the footer height and setting it in the renderer and it all started working.

Here is a sneak peak with it working with many `FlyoutItems`

![shell_footer_demo](https://user-images.githubusercontent.com/17751436/59005333-07d56280-87eb-11e9-8635-8d487dc84f28.gif)

**Comment by fakhravari** [comment]
When it is released

**Comment by douglassimaodev** [comment]
We are in XF 4.8.. next month will go to 5.0 and yet we don't have Shell.FlyoutFooter>.... any idea when/if it will be available?

**Comment by VladislavAntonyuk** [comment]
It is already available in nightly builds. So should be available in 5.0
